system/git_config.py

```diff
@@ -0,0 +1,219 @@
+  scope:
+    description:
+      - Specify which scope to read/set values from. This is required
+        when setting config values. If this is set to local, you must
+        also specify the repo parameter. It defaults to system only when
+        not using I(list_all)=yes.
+    required: false
+    choices: [ "local", "global", "system" ]
+    default: null
+  value:
+    description:
+      - When specifying the name of a single setting, supply a value to
+        set that setting to the given value.
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+# Set some settings in ~/.gitconfig
+- git_config: name=alias.ci scope=global value=commit
+- git_config: name=alias.st scope=global value=status
+
+# Or system-wide:
+- git_config: name=alias.remotev scope=system value="remote -v"
+- git_config: name=core.editor scope=global value=vim
+# scope=system is the default
+- git_config: name=alias.diffc value="diff --cached"
+- git_config: name=color.ui value=auto
+
+# Make etckeeper not complain when invoked by cron
+- git_config: name=user.email repo=/etc scope=local value="root@{{ ansible_fqdn }}"
+
+# Read individual values from git config
+- git_config: name=alias.ci scope=global
+# scope=system is also assumed when reading values, unless list_all=yes
+- git_config: name=alias.diffc
+
+# Read all values from git config
+- git_config: list_all=yes scope=global
+# When list_all=yes and no scope is specified, you get configuration from all scopes
+- git_config: list_all=yes
+# Specify a repository to include local settings
+- git_config: list_all=yes repo=/path/to/repo.git
+'''
+
+RETURN = '''
+---
+config_value:
+  description: When list_all=no and value is not set, a string containing the value of the setting in name
+  returned: success
+  type: string
+  sample: "vim"
+
+config_values:
+  description: When list_all=yes, a dict containing key/value pairs of multiple configuration settings
+  returned: success
+  type: dictionary
+  sample:
+    core.editor: "vim"
+    color.ui: "auto"
+    alias.diffc: "diff --cached"
+    alias.remotev: "remote -v"
+'''
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            list_all=dict(required=False, type='bool', default=False),
+            name=dict(type='str'),
+            repo=dict(type='path'),
+            scope=dict(required=False, type='str', choices=['local', 'global', 'system']),
+            value=dict(required=False)
+        ),
+        mutually_exclusive=[['list_all', 'name'], ['list_all', 'value']],
+        required_if=[('scope', 'local', ['repo'])],
+        required_one_of=[['list_all', 'name']],
+        supports_check_mode=True,
+    )
+    git_path = module.get_bin_path('git')
+    if not git_path:
+        module.fail_json(msg="Could not find git. Please ensure it is installed.")
+
+    params = module.params
+    # We check error message for a pattern, so we need to make sure the messages appear in the form we're expecting.
```

Comment:
That comment is false, since there is no checking of the error message as far as I see.